src/PicoClockHw/HttpRequest.cpp

```diff
@@ -0,0 +1,146 @@
+#include "HttpRequest.h"
+#include "Utils/Trace.h"
+#include "Utils/Trampoline.h"
+#include "Clock.h"
+
+
+
+HttpRequest::HttpRequest()
+{
+    m_tlsConfig = altcp_tls_create_config_client(NULL, 0);
+    MAKE_TRAMPOLINE(HttpRequest, tlsAlloc, userPtrAtBegin);
+    m_tlsAllocator.alloc = tlsAlloc;
+    m_tlsAllocator.arg = this;
+
+    // TODO: add a destructor that releases m_tlsConfig
+}
+void HttpRequest::start()
```

Comment:
By making the method generic, specific aspects of the weather feature can be moved to the class that calls this one.
```suggestion
void HttpRequest::start(const std::string &serverName, uint16_t port, const std::string &uri)
```